rewrites/src/main/scala/fix/scala213/ExplicitNonNullaryApply.scala

```diff
@@ -45,6 +45,13 @@ final class ExplicitNonNullaryApply(global: LazyValue[ScalafixGlobal])
         if !power.isJavaDefined(name) // !info.isJava
         if cond(info.signature) {
           case MethodSignature(_, List(Nil, _*), _) => true
+          case ClassSignature(_, _, _, decls) if tree.isInstanceOf[Term.ApplyType] =>
+            decls.exists { decl =>
+              decl.displayName == "apply" &&
```

Comment:
I think we can drop this one, as it wouldn't compile other, no?

Reply:
No. Given
```scala
object Props {
    def apply[A]() = ??? // (1)
    def foo() = ???          // (2)
}
```
We want fix `Props[Int]` to `Props[Int]()` but only if (1) is non-nullary.
+ `cond(decl.signature) { ... ` condition to check `decl` is a non-nullary method
+ `decl.displayName == "apply"` condition to check `decl` is (1), not another method like (2)

Reply:
That means:
+ Without `decl.displayName == "apply"` then we will incorrectly add parens to `Props[Int]` for
```scala
object Props {
    def apply[A] = ??? // nullary!
    def foo() = ??? 
}
``` 
+ Without `cond(decl.signature) { ...` then we will incorrectly add parens to `Props[Int]` for
```scala
object Props {
    def apply[A] = ???
}
```

Reply:
Ah, I see!